spectral_cube/spectral_cube.py

```diff
@@ -2028,6 +2033,77 @@ def subcube_from_regions(self, region_list, allow_empty=False):
         # that we can now crop out
         return masked_subcube.minimal_subcube(spatial_only=True)
 
+    def _velocity_freq_conversion(self, ranges, veltypes, restfreqs):
+
+        header = self.wcs.to_header()
+
+        # Obtaining rest frequency of the cube in GHz.
+        restfreq_cube = header['restfrq'] * u.Hz
+
+        CTYPE3 = header['CTYPE3']
+
+        if 'RAD' in CTYPE3:
+            veltype_cube = 'RADIO'
+        elif 'OPT' in CTYPE3:
+            veltype_cube = 'OPTICAL'
+        else:
+            veltype_cube = 'RELATIVISTIC'
+
+        def doppler_z(restfreq):
+            return [(u.GHz, u.km / u.s,
+                        lambda x: (restfreq - x) / x,
+                        lambda x: restfreq / (1 + x))]
+
+        def doppler_beta(restfreq):
+            return [(u.GHz, u.km / u.s,
+                    lambda x: si.c.to_value('km/s') * (1 - (x / restfreq) ** 2)
+                            / (1 + (x / restfreq) ** 2),
+                    lambda x: np.sqrt(
+                        (restfreq * ((x - si.c.to_value("km/s")) / (x + si.c.to_value("km/s")))))
+                     )]
+
+        def doppler_gamma(restfreq):
+            return [(u.GHz, u.km / u.s,
+                        lambda x: si.c.to_value("km/s") * (1 + (x / restfreq)**2) /
+                                  (2 * x / restfreq),
+                        lambda x: restfreq * ((x +
+                                  np.sqrt(x ** 2 - si.c.to_value("km/s") ** 2))
+                                  / si.c.to_value("km/s"))
+                     )]
```

Comment:
These are nicely done, but they are all implemented in astropy.  You can use this code:
https://github.com/radio-astro-tools/spectral-cube/blob/master/spectral_cube/spectral_axis.py#L90
to determine which to use

Reply:
I checked the code, the velocity convention `RADIO`, `RELATIVISTIC`, `OPTICAL` are handled there, but not 
these : `Z`, `BETA`, `GAMMA`

Reply:
What are z, beta, gamma?  I've never heard of these before.  Maybe they should be implemented in astropy...

Reply:
Right, `z` is redshift, `gamma` is the relativistic gamma, beta... I don't know.  These should be implemented in astropy, but in the meantime, maybe move all of the spectral-handling code to `spectral_axis`.